packages/headless/src/app/commerce-ssr-engine/common.ts

```diff
@@ -19,6 +20,12 @@ import {
   SolutionType,
 } from './types/common.js';
 
+function hasKindProperty(
```

Comment:
You could use this guard function here instead of casting:

https://github.com/coveo/ui-kit/pull/4793/files#diff-475024fc866e566edde56105ed87bd103990b0f900165642d72565b3fb169f8dR54

Reply:
I put those changes in #4783  instead and I think this is not necessary, we are looping through every controller anyway so it won't change anything